dex/order/order.go

```diff
@@ -153,23 +153,41 @@ type Order interface {
 	// Quote returns the unique integer identifier of the quote asset as defined
 	// in the asset package.
 	Quote() uint32
+
+	// IsSell indicates the the order is selling the base asset (false indicates
+	// selling the quote asset). This helps identify the asset of the backing
+	// coins returned by CoinIDs(). Note that a cancel order will return false.
+	IsSell() bool
+
+	// Coins returns the backing coins of either base or quote asset depending
+	// on IsSell.
+	//
+	// TODO!!! The orders must be updated to track the current backing coins,
+	// not just the original backing coins. This is critical for partially
+	// filled orders where each fill creates change that must then be tracked as
+	// the new backing coins.
+	CoinIDs() []CoinID
 }
 
 // zeroTime is the Unix time for a Time where IsZero() == true.
 var zeroTime = time.Time{}.Unix()
 
 // An order's ID is computed as the Blake-256 hash of the serialized order.
 func calcOrderID(order Order) OrderID {
-	stime := order.Time()
-	if stime == zeroTime {
+	sTime := order.Time()
+	if sTime == zeroTime {
 		panic("Order's ServerTime is unset")
 	}
 	return blake256.Sum256(order.Serialize())
 }
 
-// CoinID identifies value on the blockchain.
+// CoinID should be used to wrap a []byte so that it may be used as a map key.
 type CoinID []byte
 
+func (c CoinID) String() string {
+	return hex.EncodeToString(c)
+}
+
```

Comment:
I considered making a function like the following, but it would have been just for the coinlocker internals:

```go
func (c CoinID) MapKey() string {
	return string(c)
}
```